src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs

```diff
@@ -133,8 +134,13 @@ private void FileActionOccurred(object sender, TextDocumentFileActionEventArgs e
             {
                 FilePath = e.FilePath;
                 ProjectItem = dte.Solution.FindProjectItem(this.FilePath);
+
+                // Update and publish a new snapshow with the existing issues so 
+                // that the name change propagates to items in the error list
+                RefreshIssues();
```

Comment:
... and if a file name occurs we need to re-publish the data to the Error List (otherwise the issues will still show the old file name and double-clicking on an error won't navigate to the error).